### DISCUSSION TITLE
Case insensitive autocomplete

### DISCUSSION THREAD

**Discussion by gramakri** [firefly-iii/firefly-iii#8199] [Firefly III]
Hi,
I am maintainer of the Cloudron package for Firefly-III.  Recently there was a question about case sensitivity of auto completions in our [forum](https://forum.cloudron.io/topic/10525/case-sensitivity-lower-upper-case-for-suggestions/2) . The [issue](https://github.com/firefly-iii/firefly-iii/issues/8187) recommends https://docs.firefly-iii.org/firefly-iii/faq/other/#auto-complete-is-case-sensitive . There are some previous discussions at https://github.com/orgs/firefly-iii/discussions/5969 and https://github.com/orgs/firefly-iii/discussions/4775 which blame the database.

The wiki says "When the auto-complete searches in your database for entries, it may do so in a case-sensitive manner." . That is true but what exactly must the database admin do to make it case insensitive? AFAIK, it's the application that queries the database and it must use the various options that are listed in https://aws.amazon.com/blogs/database/manage-case-insensitive-data-in-postgresql/ . What should the database/platform do ? AFAIK, case insensitive collations are also created by the application in it's migration scripts.

Or is the wiki entry saying that this problem cannot be solved when using PostgreSQL? In which case, we will migrate the Cloudron package to use MySQL instead.

Thanks!

**Comment by JC5** [comment]
If I use PostgreSQL specific query code, all other database are instantly incompatible, so that's not an option. If you have some pointers for case insensitive collations for postgresql, I will see what I can do.

**Reply by gramakri** [reply]
Thanks @JC5 . I see, fair enough. What do you recommend here (not just for Cloudron package but also for new installations), should we just move to MySQL or figure a way to make PostgreSQL work?

**Reply by JC5** [reply]
I think MySQL would be the easiest, personally.
